### ISSUE TITLE
critpath -1 karma is not reflected in overall karma

### ISSUE THREAD

**Issue by kparal** [fedora-infra/bodhi#579]
In this update https://bodhi.fedoraproject.org/updates/FEDORA-2015-15804 pschindl added -1 critpath comment to it. That's a response to question "Does the system's basic functionality continue to work after this update?". If the answer is negative, it means we have a serious problem somewhere. Yet it is not reflect in global karma, which still says 0 (instead of -1).

![bodhi](https://cloud.githubusercontent.com/assets/755451/9932594/87ea7944-5d45-11e5-8185-fdf49a5a4564.png)

We don't want to autopush updates where people give -1 critpath votes. Currently it seems Bodhi ignores that.

The general karma votes ("Is the update generally functional?") and critpath votes ("Does the system's basic functionality continue to work after this update?") should both affect overall karma vote. If any of those votes is -1, the overall karma vote should be -1.

**Comment by trishnaguha** [comment]
I think we can fix this issue and #829 if we can fix #879 .
The PR is on the way https://github.com/fedora-infra/bodhi/pull/894

**Comment by kparal** [comment]
Thanks for working on this, @trishnaguha!

**Comment by trishnaguha** [comment]
#894 makes sure that Autopush will be disabled automatically whenever a critical update receives negative karma.
@kparal @bowlofeggs Sounds like we can close this issue?

**Comment by kparal** [comment]
@trishnaguha Just to clarify, does bodhi now give -1 karma even for "Does the system's basic functionality continue to work after this update?" question?

**Comment by bowlofeggs** [comment]
@trishnaguha I don't think the -1 on the critpath question will count as a negative karma, so I don't think #894 will solve this particular issue.

**Comment by trishnaguha** [comment]
#894 doesn't deal with critpath karma.

**Comment by trishnaguha** [comment]
To clarify my understanding regarding this issue, we want to make sure that Autopush is disabled if anyone gives critpath karma -1. @bowlofeggs @kparal right? 
Also we want to include critpath karma to total karma if it is negative which definitely sounds like some changes.

**Comment by kparal** [comment]
> Also we want to include critpath karma to total karma if it is negative

Yes. If the user gives either karma -1 **or** critpath -1, the overall karma should receive -1.

> we want to make sure that Autopush is disabled if anyone gives critpath karma -1

That's actually #833, also reported by me, proposed as a separate change. I'd also like to see that, yes. And this is not just about critpath -1, again it's karma -1 **or** critpath -1.

**Comment by bowlofeggs** [comment]
I think I agree with @kparal, it all sounds reasonable to me.

**Comment by hroncok** [comment]
I've noticed this and wanted to report it.

https://bodhi.fedoraproject.org/updates/FEDORA-2019-29c418b1d7

This update got 2 negative critpath karma and nothing happened. It was only when somebody gave -1 normal karma, where bodhi disabled automatic push to stable and the overall karma counter decreased.

![_](https://user-images.githubusercontent.com/2401856/64327244-a9777600-cfcb-11e9-83a1-47e6fd288ab7.png)

![_](https://user-images.githubusercontent.com/2401856/64327304-c318bd80-cfcb-11e9-9a49-a03988282124.png)

(Note that the update was manually unpushed by the submitter.)

**Comment by sebwoj** [comment]
@ryanlerch Since `karma_critpath` is deprecated:
https://github.com/fedora-infra/bodhi/blob/25cd8e7d211b3adab233c1712c6005ce1e0828a4/bodhi/server/models.py#L4209
can we close this issue?
